src/openloco/windows/townwnd.cpp

```diff
@@ -1,31 +1,863 @@
+            gfx::draw_string_494BBF(*dpi, xPos, yPos, width, colour::black, string_ids::status_town_population, &*commonFormatArgs);
+        }
+
+        // 0x00499079
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::caption:
+                    common::renameTownPrompt(self, widgetIndex);
+                    break;
+
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_town:
+                case common::widx::tab_population:
+                case common::widx::tab_company_ratings:
+                    common::switchTab(self, widgetIndex);
+                    break;
+
+                case widx::centre_on_viewport:
+                {
+                    if (self->viewports[0] == nullptr || self->saved_view.isEmpty())
+                        break;
+
+                    registers regs;
+                    if (self->saved_view.hasUnkFlag())
+                    {
+                        uint32_t townId = self->saved_view.getThingId();
+                        auto townCentre = thingmgr::get<Thing>(townId << 7);
+
+                        regs.ax = townCentre->x;
+                        regs.cx = townCentre->y;
+                        regs.dx = townCentre->z;
+                    }
+                    else
+                    {
+                        regs.ax = self->saved_view.mapX;
+                        regs.cx = self->saved_view.mapY & 0x3FFF;
+                        regs.dx = self->saved_view.surfaceZ;
+                    }
+
+                    regs.esi = (int32_t)WindowManager::getMainWindow();
+                    call(0x004C6827, regs);
+                    break;
+                }
+
+                case widx::expand_town:
+                {
+                    auto town = townmgr::get(self->number);
+
+                    const uint32_t ebx = (town->var_38 >> 3) + 5;
+                    const int16_t currentYear = current_year();
+                    int16_t tempYear = currentYear - 51;
+                    set_current_year(tempYear);
+
+                    for (uint8_t i = 8; i > 0; i--)
+                    {
+                        for (uint32_t j = ebx; j > 0; j--)
+                        {
+                            registers regs;
+                            regs.esi = (int32_t)town;
+                            regs.eax = 0xFF;
+
+                            call(0x00498116, regs);
+                            call(0x004975E0, regs);
+                        }
+
+                        tempYear += 7;
+                        set_current_year(tempYear);
+                    }
+
+                    set_current_year(currentYear);
+
+                    // Set new history entry.
+                    uint8_t historyEntry = std::min<uint8_t>(town->population / 50, 255);
```

Comment:
Just to be sure, does `town->population / 50` get clamped to an `uint8_t` before or after clamping of the value? I understand that it does't matter in this case, since it's unsigned. But would still like to know.

Reply:
Both arguments are interpreted as a value (object) of the type (class) passed. As they are rvalues, this re-interpretation takes place after division. Consider:
```
template<class T> 
const T& min(const T& a, const T& b)
{
    return (b < a) ? b : a;
}
```
(via [cppref](https://en.cppreference.com/w/cpp/algorithm/min#Possible_implementation))